kedro/framework/startup.py

```diff
@@ -86,14 +88,31 @@ def _get_project_metadata(project_path: Union[str, Path]) -> ProjectMetadata:
             f"configuration parameters."
         ) from exc
 
-    mandatory_keys = ["package_name", "project_name", "project_version"]
+    mandatory_keys = ["package_name", "project_name"]
     missing_keys = [key for key in mandatory_keys if key not in metadata_dict]
     if missing_keys:
         raise RuntimeError(f"Missing required keys {missing_keys} from '{_PYPROJECT}'.")
 
+    # Temporary solution to keep project_version backwards compatible to be removed in 0.19.0
+    if "project_version" in metadata_dict:
+        warnings.warn(
+            "project_version in pyproject.toml is deprecated, use kedro_init_version instead",
+            DeprecationWarning,
+        )
+        kedro_init_version = metadata_dict["project_version"]
+        metadata_dict["kedro_init_version"] = kedro_init_version
+    elif "kedro_init_version" in metadata_dict:
+        kedro_init_version = metadata_dict["kedro_init_version"]
+        metadata_dict["project_version"] = kedro_init_version
+    else:
+        raise RuntimeError(
+            f"Missing required key kedro_init_version from '{_PYPROJECT}'."
+        )
+
+    mandatory_keys.append("kedro_init_version")
     # check the match for major and minor version (skip patch version)
-    if metadata_dict["project_version"].split(".")[:2] != kedro_version.split(".")[:2]:
-        raise ValueError(_version_mismatch_error(metadata_dict["project_version"]))
+    if kedro_init_version.split(".")[:2] != kedro_version.split(".")[:2]:
```

Comment:
I was a bit confused when reading this logic, as I understand that `metadata_dict` will be patched based on whether certain keys exist. If this is the case, I think we can remove the variable `kedro_init_version` and use `metadata_dict["kedro_init_version"]` as this will be the final truth.